src/main/java/emissary/util/Hexl.java

```diff
@@ -14,24 +14,37 @@
     static final int COUNTER_SIZE = 4;
     static final String printable = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890!@#$%^&*()_+|~-=\\`{}[]:;<>?,./'\"";
 
+    static final char[] hexFirstCharacters = new char[256];
+    static final char[] hexSecondCharacters = new char[256];
+
+    static {
+        for (int i = 0; i <= 0xFF; i++) {
+            String s = Integer.toHexString(i & 0xff);
+            if (s.length() < 2) {
+                hexFirstCharacters[i] = '0';
+                hexSecondCharacters[i] = s.charAt(0);
+            } else {
+                hexFirstCharacters[i] = s.charAt(0);
+                hexSecondCharacters[i] = s.charAt(1);
+            }
+        }
+    }
+
     /**
      * Unformatted hex string
      */
     public static String toUnformattedHexString(byte[] data) {
-        StringBuilder output = new StringBuilder(2048);
+        final char[] characters = new char[data.length * 2];
```

Comment:
Recommend comparing performance against ```org.apache.commons.codec.binary.Hex.encodeHexString```. If the performance differences are negligible, I recommend going with the Apache commons implementation.

Reply:
The new code and the encodeHexString method are practically the same so their performance should be practically the same; therefore, I have replaced the new code with the encodeHexString method.